website/source/docs/secrets/mongodb/index.html.md

```diff
@@ -0,0 +1,386 @@
+---
+layout: "docs"
+page_title: "Secret Backend: mongodb"
+sidebar_current: "docs-secrets-mongodb"
+description: |-
+  The mongodb secret backend for Vault generates database credentials to access MongoDB.
+---
+
+# MongoDB Secret Backend
+
+Name: `mongodb`
+
+The `mongodb` secret backend for Vault generates MongoDB database credentials
+dynamically based on configured roles. This means that services that need
+to access a MongoDB database no longer need to hard-code credentials: they
+can request them from Vault and use Vault's leasing mechanism to more easily
+roll them.
+
+Additionally, it introduces a new ability: with every service accessing
+the database with unique credentials, it makes auditing much easier when
+questionable data access is discovered: you can track it down to the specific
+instance of a service based on the MongoDB username.
+
+Vault makes use of its own internal revocation system to ensure that users
+become invalid within a reasonable time of the lease expiring.
+
+This page will show a quick start for this backend. For detailed documentation
+on every path, use `vault path-help` after mounting the backend.
+
+## Quick Start
+
+The first step to using the mongodb backend is to mount it.
+Unlike the `generic` backend, the `mongodb` backend is not mounted by default.
+
+```
+$ vault mount mongodb
+Successfully mounted 'mongodb' at 'mongodb'!
+```
+
+Next, we must tell Vault how to connect to MongoDB. This is done by providing
+a standard connection string (URI):
+
+```
+$ vault write mongodb/config/connection uri="mongodb://admin:Password!@mongodb.acme.com:27017/admin?ssl=true"
```

Comment:
Well, that ultimately is up to us, because we have to handle establishing the SSL connection; `mgo` doesn't include any direct support. So it comes down to this chunk of code in `util.go`:

```
case "ssl":
	if value == "true" {
		info.DialServer = func(addr *mgo.ServerAddr) (net.Conn, error) {
			return tls.Dial("tcp", addr.String(), &tls.Config{})
		}
	}
```